api/v1alpha1/eventing_types.go

```diff
@@ -74,72 +81,71 @@ type EventingList struct {
 // Backend defines eventing backend.
 type Backend struct {
 	// Type defines which backend to use. The value is either `EventMesh`, or `NATS`.
-	// +kubebuilder:validation:Enum=EventMesh;NATS
+	// +kubebuilder:default:="NATS"
+	// +kubebuilder:validation:XValidation:rule="self=='NATS' || self=='EventMesh'", message="storage type can only be set to NATS or EventMesh"
 	Type string `json:"type"`
 
 	// Config defines configuration for eventing backend.
-	// +optional
+	// +kubebuilder:default:={natsStreamStorageType:"File", natsStreamReplicas:3, natsStreamMaxSize:"700Mi", natsMaxMsgsPerTopic:1000000}
 	Config BackendConfig `json:"config"`
 }
 
 // BackendConfig defines configuration for eventing backend.
 type BackendConfig struct {
-	// NatsStorageType defines the storage type for stream data.
-	// +optional
-	// +kubebuilder:validation:Enum=File;Memory
-	NATSStorageType string `json:"natsStorageType"`
-
-	// NatsStorageSize defines the storage size for stream data.
+	// NATSStreamStorageType defines the storage type for stream data.
+	// +kubebuilder:default:="File"
+	// +kubebuilder:validation:XValidation:rule="self=='File' || self=='Memory'", message="storage type can only be set to File or Memory"
 	// +optional
-	NATSStorageSize resource.Quantity `json:"natsStorageSize"`
+	NATSStreamStorageType string `json:"natsStreamStorageType"`
 
-	// NatsStreamReplicas defines the number of replicas for stream.
+	// NATSStreamReplicas defines the number of replicas for stream.
+	// +kubebuilder:default:=3
 	// +optional
 	NATSStreamReplicas int `json:"natsStreamReplicas"`
 
-	// MaxStreamSize defines the maximum storage size for stream data.
+	// NATSStreamMaxSize defines the maximum storage size for stream data.
+	// +kubebuilder:default:="700Mi"
 	// +optional
-	MaxStreamSize resource.Quantity `json:"maxStreamSize"`
+	NATSStreamMaxSize resource.Quantity `json:"natsStreamMaxSize"`
 
-	// MaxMsgsPerTopic limits how many messages in the NATS stream to retain per subject.
+	// NATSMaxMsgsPerTopic limits how many messages in the NATS stream to retain per subject.
+	// +kubebuilder:default:=1000000
 	// +optional
-	MaxMsgsPerTopic int `json:"maxMsgsPerTopic"`
+	NATSMaxMsgsPerTopic int `json:"natsMaxMsgsPerTopic"`
 
 	// EventMeshSecret defines the namespaced name of K8s Secret containing EventMesh credentials. The format of name is "namespace/name".
 	// +optional
 	EventMeshSecret string `json:"eventMeshSecret"`
```

Comment:
I believe we should validate the `namespace/name` secret format.

Reply:
Done and added test case.